aya/src/programs/cgroup_skb.rs

```diff
@@ -103,13 +108,60 @@ impl CgroupSkb {
                 }
             })?;
 
-            Ok(self
-                .data
-                .link(ProgAttachLink::new(prog_fd, cgroup_fd, attach_type)))
+            self.data
+                .links
+                .insert(CgroupSkbLink(CgroupSkbLinkInner::ProgAttach(
+                    ProgAttachLink::new(prog_fd, cgroup_fd, attach_type),
+                )))
+        }
+    }
+
+    /// Detaches the program.
+    ///
+    /// See [CgroupSkb::attach].
+    pub fn detach(&mut self, link_id: CgroupSkbLinkId) -> Result<(), ProgramError> {
```

Comment:
Could we accept a **reference** to `*LinkId` here? That would be convenient for storing link_ids in hash maps, where `get` and `get_mut` are giving a reference rather than owned element.

For example:

```
fn attach_program(bpf: &mut Bpf, my_bpf_links: &mut HashMap<String, UProbeLinkId>, my_id: String) -> Result<(), anyhow::Error> {
    let program: &mut UProbe = bpf.program_mut("my_program").unwrap().try_info()?;
    program.load()?;
    let link_id = program.attach(Some("getaddrinfo"), 0, "libc", None)?;

    my_bpf_links.insert(my_id, link_id);

    Ok(())
}

fn detach_program(my_bpf_links: &mut HashMap<String, UProbeLinkId>, my_id: String) -> Result<(), anyhow::Error> {
    let program: &mut UProbe = bpf.program_mut("my_program").unwrap().try_info()?;
    match my_bpf_links.get(my_id) {
        Some(link_id) => program.detach(link)?, // Here we get &UProbeLinkId !!!
        None => return Err(anyhow!("missing link")),
    }

    Ok(())
}
```

Currently it's not possible.

Reply:
Or hmm... actually using `remove` on HashMap always solves the problem. But anyway, there might be people wanting to use a reference for some other reason.

Reply:
Nope taking self by value is very much intended: after detach a link becomes unusable (can't detach twice), and moving self ensures that trying to detach twice would fail at compile time.

Reply:
OK, good point, then seems like `remove` is the only way to go with HashMaps on the userspace side.